timelock-impl/src/test/java/com/palantir/atlasdb/backup/AtlasRestoreResourceTest.java

```diff
@@ -53,8 +57,18 @@
     @Mock
     private AsyncTimelockService otherTimelock;
 
-    private final AtlasRestoreResource atlasRestoreResource =
-            new AtlasRestoreResource(TARGETER, str -> str.equals("test") ? mockTimelock : otherTimelock);
+    private final AtlasRestoreResource atlasRestoreResource = new AtlasRestoreResource(
+            () -> BEARER_TOKEN, TARGETER, str -> str.equals("test") ? mockTimelock : otherTimelock);
+
+    @Test
+    public void throwsIfWrongAuthHeaderIsProvided() {
+        CompletedBackup completedBackup = completedBackup();
+        AuthHeader wrongHeader = AuthHeader.of(BearerToken.valueOf("imposter"));
+        CompleteRestoreRequest request = CompleteRestoreRequest.of(ImmutableSet.of(completedBackup));
+        assertThatServiceExceptionThrownBy(
+                        () -> AtlasFutures.getUnchecked(atlasRestoreResource.completeRestore(wrongHeader, request)))
+                .hasType(ErrorType.PERMISSION_DENIED);
```

Comment:
Again, so we have an assert on when this does not throw?

Reply:
I guess I could have something like `assertThat(...).doesNotThrow();` to make it really explicit?